geode-core/src/main/java/org/apache/geode/management/internal/api/LocatorClusterManagementService.java

```diff
@@ -288,58 +289,30 @@ public LocatorClusterManagementService(ConfigurationPersistenceService persisten
       resultList.add(filter);
     } else {
       ConfigurationManager<T> manager = getConfigurationManager(filter);
-      // gather elements on all the groups, consolidate the group information and then do the filter
-      // so that when we filter by a specific group, we still show that a particular element might
-      // also belong to another group.
-      for (String group : persistenceService.getGroups()) {
-        CacheConfig currentPersistedConfig = persistenceService.getCacheConfig(group, true);
-        List<T> listInGroup = manager.list(filter, currentPersistedConfig);
-        for (T element : listInGroup) {
-          element.setGroup(group);
-          resultList.add(element);
-        }
-      }
-
-      // if empty result, return immediately
-      if (resultList.size() == 0) {
-        return result;
+      Set<String> groups;
+      if (StringUtils.isNotBlank(filter.getGroup())) {
+        groups = Collections.singleton(filter.getGroup());
+      } else {
+        groups = persistenceService.getGroups();
       }
 
-      // right now the list contains [{regionA, group1}, {regionA, group2}...], we need to
-      // consolidate the list into [{regionA, [group1, group2]}
-      List<T> consolidatedResultList = new ArrayList<>();
-      for (T element : resultList) {
-        int index = consolidatedResultList.indexOf(element);
-        if (index >= 0) {
-          T exist = consolidatedResultList.get(index);
-          exist.addGroup(element.getGroup());
-        } else {
-          consolidatedResultList.add(element);
+      for (String group : groups) {
+        CacheConfig currentPersistedConfig = persistenceService.getCacheConfig(group, true);
```

Comment:
group should be canonicalized before passing to getCacheConfig, otherwise filtering for "ClUsTeR" does not match anything

Reply:
you mean group comparison should ignore case?

Reply:
fixed